troubleshooting/devtools.md

```diff
@@ -0,0 +1,188 @@
+# Dev Tools Guide
+
+## VS Code Remote Setup
+**VSCode Remote**  
+After installing [VS Code](https://code.visualstudio.com/Download), the following tutorial can be used to connect Visual Studio Code to a remote Linux system or server. <https://code.visualstudio.com/docs/remote/ssh>
+
+Most commonly Amazon Linux 2 EC2 instances are used and can be setup using the following [guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html)
+
+[C5 EC2 instances](https://aws.amazon.com/ec2/instance-types/c5/) work well for standard development. If within budget, c5.2xLarge works great.
+
+**Fluent Bit**  
+SSH to the remote instance and clone Fluent Bit's source code and build with instructions from the [Developer Guide](https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#testing)
+
+In VS Code, connect to the remote instance and open the Fluent Bit repo. This will be the workspace we will setup.
+
+**Extensions**  
+Install the following Visual Studio Code extensions on the remote instance by clicking the extension icon on the left VS Code icon bar.
+- C/C++
+- C/C++ Extension Pack
+- CMake Tools
+- CMake
+- GitLens (helpful)
+
+**Configuration**  
+Copy the `aws-for-fluent-bit/troubleshooting/resources/vscode` resources folder to your Fluent Bit repository. Rename the folder from vscode to .vscode.
+```
+> tree ./fluent-bit/.vscode/
+├── external-plugins
+│   └── Readme.md
+├── fluent-bit-config
+│   ├── fluent-bit-cloudwatch_logs.conf
+│   ├── fluent-bit.conf
+│   ├── fluent-bit-go.conf
+│   ├── fluent-bit-kinesis_firehose.conf
+│   ├── fluent-bit-kinesis_streams.conf
+│   └── fluent-bit-s3.conf
+├── launch.json
+├── scripts
+│   ├── list-fluent-bit-plugin-args.sh
+│   ├── rebuild-external.sh
+│   └── rebuild.sh
+├── settings.json
+└── tasks.json
+```
+This .vscode folder does several things
+1. Sets style guides
+    - Indent style
+    - Ruler set to 90 characters from left
+2. Adds build scripts used by VS Code debugging
+3. Adds build tasks used by VS Code debugging
+4. Adds launch configuration `launch.json` which sets up vscode to build and run Fluent Bit
+5. External plugins folder, to auto build, attach, and run Go plugins if needed
+6. Fluent Bit config files that are connected to the Fluent Bit launch tasks
+
+If you wish to develop aws-for-fluent-bit go plugins, we recommend you follow the readme in the external plugins folder to configure these repositories.
```

Comment:
nit: change readme to full name README.md and add link to aws-for-fluent-bit go plugins: https://github.com/aws/aws-for-fluent-bit#plugins

Reply:
Actually I am referring to another readme, the one included in the external-plugins folder. Changed to refer to that doc.